schema/mutation.graphql

```diff
@@ -5,72 +5,72 @@ module.exports = `
     # removeMultipleMembers(data: MultipleUsersAndOrganizationInput!) : Organization!
 
-    adminRemovePost(organizationId: ID!, postId:ID!):Post!
-    adminRemoveEvent(eventId: ID!): Event!
-    adminRemoveGroup(groupId:ID!):Message!
+    adminRemovePost(organizationId: ID!, postId:ID!):Post! @auth(requires: Admin)
+    adminRemoveEvent(eventId: ID!): Event! @auth(requires: Admin)
+    adminRemoveGroup(groupId:ID!):Message! @auth(requires: Admin)
 
-    createPost(data: PostInput!, file: Upload): Post
-    removePost(id:ID!): Post
-    likePost(id:ID!): Post
-    unlikePost(id:ID!): Post
+    createPost(data: PostInput!, file: Upload): Post @auth
+    removePost(id:ID!): Post @auth
+    likePost(id:ID!): Post @auth
+    unlikePost(id:ID!): Post @auth
 
-    createComment(postId:ID!, data: CommentInput!): Comment
-    removeComment(id:ID!): Comment
-    likeComment(id:ID!): Comment
-    unlikeComment(id:ID!): Comment
+    createComment(postId:ID!, data: CommentInput!): Comment @auth
+    removeComment(id:ID!): Comment @auth
+    likeComment(id:ID!): Comment @auth
+    unlikeComment(id:ID!): Comment @auth
 
-    createTask(data: TaskInput, eventId: ID!): Task!
-    updateTask(id:ID!, data: UpdateTaskInput) : Task
-    removeTask(id: ID!) : Task
+    createTask(data: TaskInput, eventId: ID!): Task! @auth
+    updateTask(id:ID!, data: UpdateTaskInput) : Task @auth
+    removeTask(id: ID!) : Task @auth
 
-    createGroup(data: GroupInput!): Group!
+    createGroup(data: GroupInput!): Group! @auth
 
-    sendMembershipRequest(organizationId: ID!) : MembershipRequest!
-    acceptMembershipRequest(membershipRequestId: ID!): MembershipRequest!
-    rejectMembershipRequest(membershipRequestId: ID!): MembershipRequest!
-    cancelMembershipRequest(membershipRequestId: ID!) : MembershipRequest!
+    sendMembershipRequest(organizationId: ID!) : MembershipRequest! @auth
+    acceptMembershipRequest(membershipRequestId: ID!): MembershipRequest! @auth(requires: Admin)
+    rejectMembershipRequest(membershipRequestId: ID!): MembershipRequest! @auth(requires: Admin)
+    cancelMembershipRequest(membershipRequestId: ID!) : MembershipRequest! @auth(requires: Admin)
 
-    blockUser(organizationId: ID!, userId: ID!) : User!
-    unblockUser(organizationId: ID!, userId:ID!) : User!
+    blockUser(organizationId: ID!, userId: ID!) : User! @auth(requires: Admin)
+    unblockUser(organizationId: ID!, userId:ID!) : User! @auth(requires: Admin)
 
-    addUserImage(file: Upload!): User!
-    removeUserImage: User!
-    addOrganizationImage(file: Upload!, organizationId: String!): Organization!
-    removeOrganizationImage(organizationId: String!): Organization!
+    addUserImage(file: Upload!): User! @auth
+    removeUserImage: User! @auth
+    addOrganizationImage(file: Upload!, organizationId: String!): Organization! @auth(requires: Admin)
+    removeOrganizationImage(organizationId: String!): Organization! @auth(requires: Admin)
```

Comment:
@aniketbiswas21  

Can we add multiple schema derivatives to the same mutation?

Reply:
@impiyush83 Yes sir we can.

Reply:
@aniketbiswas21  Cool. It will be better to segregate auth into 2 directives such as:

1.  authenticate 
2. authorize 

Reason :  

In the future, if we want to add more granular RBAC we can make a change in authorize directive.

Reply:
> @aniketbiswas21 Cool. It will be better to segregate auth into 2 directives such as:
> 
> 1. authenticate
> 2. authorize
> 
> Reason :
> 
> In the future, if we want to add more granular RBAC we can make a change in authorize directive.

@impiyush83 @sumitra19jha Should I go ahead with the above changes?

Reply:
@aniketbiswas21 We can go ahead only with the authentication only. We can create a AuthorizationDirective later as a separate PR

@sumitra19jha Do you think the same?

Reply:
Where do we need Authorization?

Reply:
If there is any resolver or service where Authorization is needed then we will add it else we will deal with them when needed in future.